backend/src/main/java/c/team/account/UserAccountController.java

```diff
@@ -0,0 +1,40 @@
+package c.team.account;
+
+import c.team.account.model.LoginRequest;
+import c.team.account.model.RegisterAccountRequest;
+import c.team.account.model.UserAccount;
+import c.team.token.TokenService;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@AllArgsConstructor(onConstructor = @__(@Autowired))
```

Comment:
To rozumiem, że dodaje adnotację @Autowired do konstruktora, ale ciężko mi było z neta zrozumieć co robi ta adnotacja - mógłbyś wyjaśnić?

Reply:
Spring to generalnie takie narzędzie do robienia dependency injection. Jak mamy jakies objekty, którymi zarządza spring (tak zwane beany, między innymi klasy z adnotacją `@Service`, `@Repository`), to on zajmuje się wstrzykiwaniem ich w miejsca, gdzie powinny się znaleźć. Takie miejsca oznaczamy adnotacją `@Autowired`. W tym przypadku mówimy springowi żeby dodał do `UserController` beany `UserAccountService ` i `TokenService`.